### PR TITLE
MemoryAccessRight and Cache maintenance for SDP (SWG-186)

### PR DESCRIPTION
**Pull request by etienne-lms**
New TEE Internal API extension: on TEE_CheckMemoryAccesRights(). 
[description here](https://docs.google.com/document/d/1QaCt3GgWwsQBTZ5xSHkYoavipgNC27cCkjiOrnOFE9s).

Fix on TA cache API.

### CODE REVIEW

File: lib/libutee/include/tee_internal_api_extensions.h [hunk 1]
```diff
@@ -32,11 +32,8 @@
 #include <trace.h>
 #include <stdio.h>
 #include <tee_api_types.h>
+#include <tee_api_defines_extensions.h>
```
Comment (etienne-lms) [inline lib/libutee/include/tee_internal_api_extensions.h]: Move up 1 line

**Comment by etienne-lms** [comment]
fix from comments.
added a patch to fix TA private memory tests.
rebased.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
tag applied. waiting for travis. should not complain (tested on qemu)

**Comment by jbech-linaro** [comment]
@etienne-lms I merged your other PR a few hours ago, so unfortunately this needs another rebase. Would you mind do that and I can merge it right away?

**Comment by etienne-lms** [comment]
rebased

**Pull request merged**